lib/verifier/post_zephir_verifier.rb

```diff
@@ -0,0 +1,195 @@
+# frozen_string_literal: true
+
+require "zlib"
+require_relative "../verifier"
+require_relative "../derivatives"
+
+# Verifies that post_zephir workflow stage did what it was supposed to.
+
+# TODO: document and verify the files written by monthly process.
+# They should be mostly the same but need to be accounted for.
+
+module PostZephirProcessing
+  class PostZephirVerifier < Verifier
+    attr_reader :current_date
+
+    def run_for_date(date:)
+      @current_date = date
+      verify_catalog_archive
+      verify_catalog_prep
+      verify_dollar_dup
+      verify_ingest_bibrecords
+      verify_rights
+      verify_zephir_data
+    end
+
+    # Frequency: ALL
+    # Files: CATALOG_ARCHIVE/zephir_upd_YYYYMMDD.json.gz
+    #   and potentially CATALOG_ARCHIVE/zephir_full_YYYYMMDD_vufind.json.gz
+    # Contents: ndj file with one catalog record per line
+    # Verify:
+    #   readable
+    #   line count must be the same as input JSON
+    def verify_catalog_archive(date: current_date)
+      zephir_update_derivative_params = {
+        location: :CATALOG_ARCHIVE,
+        name: "zephir_upd_YYYYMMDD.json.gz",
+        date: date
+      }
+      zephir_update_path = self.class.dated_derivative(**zephir_update_derivative_params)
+      verify_file(path: zephir_update_path)
+      verify_parseable_ndj(path: zephir_update_path)
+
+      if date.last_of_month?
+        zephir_full_derivative_params = {
```

Comment:
Maybe there's a missing class here around `Derivative`?

Reply:
While we do need to pass location, name, date around a bunch, but it may be useful to have something that can return these three parameters along with the path, and knows whether it's a so-called "full".